### ISSUE TITLE
Upgrade SNOPT binding

### ISSUE THREAD

**Issue by hongkai-dai** [RobotLocomotion/drake#9092]
> Please fill out all quoted sections (`> `) that need a response; feel free
to remove the quoted sections that do not need a response.

> Before you post:

> * Have you reviewed the documentation on
[Getting Help](http://drake.mit.edu/getting_help.html)?

Yes
> * Have you reviewed existing GitHub issues and StackOverflow posts?

Yes
> * Have you rerun `install_prereqs.sh` on your current commit of `drake`?

Yes

## Description

> Please provide a short description: expected behavior, and what actually
happens. If this is an enhancement request, please provide a concrete use case.

The goal is to be able to run multiple SNOPT calls in parallel. There are several steps we need/can to do.
1. Upgrade SNOPT to newer versions.
   I believe we are using SNOPT 7.2.12 https://github.com/RobotLocomotion/snopt/blob/master/README.WhatsNew. We should upgrade SNOPT to newer versions, like 7.6.
2. We currently use SNOPT C interface, together with f2c (fortran to C). Should we consider to use SNOPT c++ interface? Will the C++ interface support / be better suited with running multiple SNOPT?

@jwnimmer-tri @avalenzu @RussTedrake 

There are related/overlapping issues #7984 #4893. We should investigate using https://github.com/snopt/snopt-interface/blob/master/include/snopt.h interface, rather than the f2c interface.

**Comment by hongkai-dai** [comment]
@RussTedrake , yes. As it turns out we are already on SNOPT 7.6.1 (not the master branch of RobotLocomotion/snopt repo where I looked, but on a separate branch). So we only need to do 2 here.

**Comment by hongkai-dai** [comment]
I tried to change all the snopt calls in `snopt_solver.cc` with the API in `snopt-interface`, but I couldn't get the build system to work. As @jwnimmer-tri and I discussed over the email, we will need to modify the build file in order to use snopt-interface. We will do this in two steps:

1. Jeremy or Jamie will use gfortran compiler to compile snopt.
2. I will port `snopt_solver` on top of the step 1.
